pkg/controller/operatorsource/operatorsource_controller_test.go

```diff
@@ -0,0 +1,201 @@
+package operatorsource_test
```

Comment:
I assume these are the unit tests for operatorsource/handler.go? I can't review what has changed, it is showing up as a new file.

Reply:
They are similar tests in content, but they all had to be rewritten because the logic changed significantly. Because of that, I just created new files.

Reply:
I prefer these tests getting re-written using the new fake client

Reply:
If `pkg/operatorsource` is left intact then the unit test(s) here would mock the `Handler` interface of `pkg/operatorsource` package. The advantage is that you will end up with one (at best two) unit test(s) and each will be very concise since you do not have to worry about dealing with the internals of the `Handle` function.

Reply:
This is updated based on our meeting